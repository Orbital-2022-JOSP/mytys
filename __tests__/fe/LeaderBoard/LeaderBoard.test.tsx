import { render } from '@testing-library/react';
import { LeaderBoard } from '../../../components/Leaderboard/Leaderboard';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The LeaderBoard Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<LeaderBoard />);

            expect(container).toMatchSnapshot()
        })
})

