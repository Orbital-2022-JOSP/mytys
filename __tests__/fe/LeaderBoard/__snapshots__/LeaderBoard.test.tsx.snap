// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LeaderBoard Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <div
      class="max-w-xl mb-10 md:mx-auto sm:text-center lg:max-w-2xl md:mb-12"
    >
      <div>
        <p
          class="inline-block px-3 py-px mb-4 text-xs font-semibold tracking-wider text-teal-900 uppercase rounded-full bg-teal-accent-400"
        >
          WEEKLY
        </p>
      </div>
      <h2
        class="max-w-lg mb-6 font-sans text-3xl font-bold leading-none tracking-tight text-gray-900 sm:text-4xl md:mx-auto"
      >
        <span
          class="relative inline-block"
        >
          <svg
            class="absolute top-0 left-0 z-0 hidden w-32 -mt-8 -ml-20 text-blue-gray-100 lg:w-32 lg:-ml-28 lg:-mt-10 sm:block"
            fill="currentColor"
            viewBox="0 0 52 24"
          >
            <defs>
              <pattern
                height=".30"
                id="7e5e8ff8-1960-4094-a63a-2a0c0f922d69"
                width=".135"
                x="0"
                y="0"
              >
                <circle
                  cx="1"
                  cy="1"
                  r=".7"
                />
              </pattern>
            </defs>
            <rect
              fill="url(#7e5e8ff8-1960-4094-a63a-2a0c0f922d69)"
              height="24"
              width="52"
            />
          </svg>
          <span
            class="relative"
          >
            LEADERBOARD
          </span>
        </span>
      </h2>
      <p
        class="text-base text-gray-700 md:text-lg"
      >
        Here are our top few performers of the week!
      </p>
    </div>
    <div
      class="grid max-w-md gap-10 row-gap-10 lg:max-w-screen-lg sm:row-gap-10 lg:grid-cols-3 xl:max-w-screen-lg sm:mx-auto lg:hidden"
    >
      <div
        class="flex flex-col justify-between p-8 transition-shadow duration-300 bg-white border rounded shadow-sm sm:items-center hover:shadow lg:scale-110"
      >
        <div
          class="absolute inset-x-0 lg:top-0 flex justify-center -mt-11 lg:-mt-3"
        >
          <div
            class="inline-block px-3 py-1 text-xs font-medium tracking-wider text-white uppercase rounded bg-light-blue-500"
          >
            Leader
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="text-lg font-semibold"
          >
            1st
          </div>
          <div
            class="flex items-center justify-center mt-2"
          >
            <div
              class="mr-1 text-4xl font-bold text-light-blue-500"
            >
              Josh
            </div>
          </div>
          <div
            class="mt-2 space-y-3"
          >
            <div
              class="text-gray-700"
            >
              2109
               points
            </div>
          </div>
        </div>
        <div
          class="w-full px-4 mt-3"
        >
          <ul
            class="mb-4 -ml-1 space-y-2"
          >
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Chemistry
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Math
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="flex flex-col justify-between p-8 transition-shadow duration-300 bg-white border rounded shadow-sm sm:items-center hover:shadow"
      >
        <div
          class="text-center"
        >
          <div
            class="text-lg font-semibold"
          >
            2nd
          </div>
          <div
            class="flex items-center justify-center mt-2"
          >
            <div
              class="mr-1 text-4xl font-bold text-light-blue-500"
            >
              John
            </div>
          </div>
          <div
            class="mt-2 space-y-3"
          >
            <div
              class="text-gray-700"
            >
              2000
               points
            </div>
          </div>
        </div>
        <div
          class="w-full px-4 mt-3"
        >
          <ul
            class="mb-4 -ml-1 space-y-2"
          >
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Chemistry
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Math
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Physics
            </li>
          </ul>
        </div>
      </div>
      <div
        class="flex flex-col justify-between p-8 transition-shadow duration-300 bg-white border rounded shadow-sm sm:items-center hover:shadow lg:scale-90"
      >
        <div
          class="text-center"
        >
          <div
            class="text-lg font-semibold"
          >
            3rd
          </div>
          <div
            class="flex items-center justify-center mt-2"
          >
            <div
              class="mr-1 text-4xl font-bold text-light-blue-500"
            >
              Sam
            </div>
          </div>
          <div
            class="mt-2 space-y-3"
          >
            <div
              class="text-gray-700"
            >
              1809
               points
            </div>
          </div>
        </div>
        <div
          class="w-full px-4 mt-3"
        >
          <ul
            class="mb-4 -ml-1 space-y-2"
          >
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Chemistry
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Math
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Physics
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="hidden lg:grid max-w-md gap-10 row-gap-10 lg:max-w-screen-lg sm:row-gap-10 lg:grid-cols-3 xl:max-w-screen-lg sm:mx-auto"
    >
      <div
        class="flex flex-col justify-between p-8 transition-shadow duration-300 bg-white border rounded shadow-sm sm:items-center hover:shadow"
      >
        <div
          class="text-center"
        >
          <div
            class="text-lg font-semibold"
          >
            2nd
          </div>
          <div
            class="flex items-center justify-center mt-2"
          >
            <div
              class="mr-1 text-4xl font-bold text-light-blue-500"
            >
              John
            </div>
          </div>
          <div
            class="mt-2 space-y-3"
          >
            <div
              class="text-gray-700"
            >
              2000
               points
            </div>
          </div>
        </div>
        <div
          class="w-full px-4 mt-3"
        >
          <ul
            class="mb-4 -ml-1 space-y-2"
          >
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Chemistry
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Math
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Physics
            </li>
          </ul>
        </div>
      </div>
      <div
        class="flex flex-col justify-between p-8 transition-shadow duration-300 bg-white border rounded shadow-sm sm:items-center hover:shadow lg:scale-110"
      >
        <div
          class="absolute inset-x-0 lg:top-0 flex justify-center -mt-11 lg:-mt-3"
        >
          <div
            class="inline-block px-3 py-1 text-xs font-medium tracking-wider text-white uppercase rounded bg-light-blue-500"
          >
            Leader
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="text-lg font-semibold"
          >
            1st
          </div>
          <div
            class="flex items-center justify-center mt-2"
          >
            <div
              class="mr-1 text-4xl font-bold text-light-blue-500"
            >
              Josh
            </div>
          </div>
          <div
            class="mt-2 space-y-3"
          >
            <div
              class="text-gray-700"
            >
              2109
               points
            </div>
          </div>
        </div>
        <div
          class="w-full px-4 mt-3"
        >
          <ul
            class="mb-4 -ml-1 space-y-2"
          >
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Chemistry
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Math
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="flex flex-col justify-between p-8 transition-shadow duration-300 bg-white border rounded shadow-sm sm:items-center hover:shadow lg:scale-90"
      >
        <div
          class="text-center"
        >
          <div
            class="text-lg font-semibold"
          >
            3rd
          </div>
          <div
            class="flex items-center justify-center mt-2"
          >
            <div
              class="mr-1 text-4xl font-bold text-light-blue-500"
            >
              Sam
            </div>
          </div>
          <div
            class="mt-2 space-y-3"
          >
            <div
              class="text-gray-700"
            >
              1809
               points
            </div>
          </div>
        </div>
        <div
          class="w-full px-4 mt-3"
        >
          <ul
            class="mb-4 -ml-1 space-y-2"
          >
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Chemistry
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Math
            </li>
            <li
              class="flex items-start"
            >
              <span
                class="mr-1"
              >
                <svg
                  class="w-5 h-5 mt-px text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 52 52"
                >
                  <polygon
                    fill="none"
                    points="29 13 14 29 25 29 23 39 38 23 27 23"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </span>
              Physics
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="max-w-lg space-y-6 sm:mx-auto lg:max-w-xl mt-16"
    >
      <div
        class="flex items-center p-2 transition-colors duration-200 border rounded shadow group hover:bg-light-blue-500 hover:border-light-blue-800 bg-white"
      >
        <div
          class="mr-2"
        >
          <svg
            class="w-6 h-6 transition-colors duration-200 text-light-blue-500 group-hover:text-white sm:w-8 sm:h-8"
            stroke="currentColor"
            viewBox="0 0 52 52"
          >
            <polygon
              fill="none"
              points="29 13 14 29 25 29 23 39 38 23 27 23"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
            />
          </svg>
        </div>
        <span
          class="text-gray-800 transition-colors duration-200 group-hover:text-white"
        >
          Try to participate actively in the quizzes to get a chance to be on the leaderboard!
        </span>
      </div>
    </div>
  </div>
</div>
`;
